pkg/kube/client.go

```diff
@@ -608,7 +613,8 @@ func (c *Client) AsVersionedObject(obj runtime.Object) (runtime.Object, error) {
 		return nil, err
 	}
 	versions := &runtime.VersionedObjects{}
-	err = runtime.DecodeInto(c.Decoder(true), json, versions)
+	decoder := unstructured.UnstructuredJSONScheme
```

Comment:
What's the difference here between an UnstructuredJSONScheme and kubectl's [`scheme.DefaultJSONEncoder()`](https://github.com/kubernetes/kubernetes/blob/c5d4a032d73f90f286fde1a7ca3e782538c47f58/pkg/kubectl/scheme/scheme.go#L51-L54), which appears to use a LegacyJSONEncoder? Is it that the UnstructuredJSONScheme is primarily used in APIMachinery, and the LegacyEncoder is client-side?

Reply:
UnstructuredJSONScheme is capable of converting JSON data into the Unstructured type, which can be used for generic access to objects without a predefined scheme.

LegacyJSONEncoder is type aware.

AFAIK

Reply:
Right, and (for my own understanding) so far we've been [building resources from unstructured data types](https://github.com/kubernetes/helm/blob/master/pkg/kube/client.go#L99), so it makes sense to continue to use the UnstructuredJSONScheme decoder rather than what kubectl uses. Got it. :+1: